src/client.ts

```diff
@@ -55,8 +55,37 @@ export default class UnleashClient extends EventEmitter {
     }
   }
 
+  isParentDependencySatisfied(feature: FeatureInterface | undefined, context: Context) {
+    if (!feature?.name) {
+      return true;
+    }
+
+    if (feature.dependencies && feature.dependencies.length) {
+      return feature.dependencies.every(parent => {
+        const parentFeature = this.repository.getToggle(parent.feature);
+
+        if (parentFeature?.dependencies?.length) {
```

Comment:
Can we make it more readable, Does this mean length exists or length === 0?

Reply:
nope, it should also cater for undefined dependencies. Setting === 0 breaks the spec

Reply:
move it to precondition check though